src/collector.ts

```diff
@@ -88,22 +91,62 @@ class NaivePyParser {
 class ReqDependencyCollector implements IDependencyCollector {
     constructor(public classes: Array<string> = ["dependencies"]) {}
 
-    async collect(contents: string): Promise<Array<IDependency>> {
+    async collect(contents: string, manifestFile: string): Promise<Array<IDependency>> {
         let parser = new NaivePyParser(contents);
         return parser.parse();
     }
 
 }
 
 class NaiveGomodParser {
-    constructor(contents: string) {
-        this.dependencies = NaiveGomodParser.parseDependencies(contents);
+    constructor(contents: string, manifestFile: string) {
+        this.dependencies = NaiveGomodParser.parseDependencies(contents, manifestFile);
     }
 
     dependencies: Array<IDependency>;
 
-    static parseDependencies(contents:string): Array<IDependency> {
+    static parseDependencies(contents:string, manifestFile: string): Array<IDependency> {
         const gomod = contents.split("\n");
+
+        const vscodeRootpath = manifestFile.replace("file://", "").replace("/go.mod", "")
+        const targetDir = paths.join(vscodeRootpath, 'target');
+        const goImportsFilePath = paths.join(targetDir, 'goimports.txt');
+        
+        // Create target folder if not exists.
+        if (!fs.existsSync(targetDir)) {
+            fs.mkdirSync(targetDir);
+        }
+
+        const cmd: string = [
+            `cd`,
+            `"${vscodeRootpath}" &&`,
+            `go`,
+            `list`,
+            `-f`,
+            `"{{.Imports}}"`,
+            `./...`,
+            `>`,
+            `"${goImportsFilePath}"`,
+        ].join(' ');
+        
+        execSync(
+            cmd,
+            { maxBuffer: 1024 * 1200 });
+        
+        const gopackages = fs.readFileSync(goImportsFilePath).toString().split("\n");
```

Comment:
This is first time we doing an extensive work from LSP. Please use async version of exec and use pipe to read cmd output instead of redirecting to file.

Reply:
This function is called inside async function, I thinks its a over kill to have async again here.
Reading from a pipe can be done instead of a file.

Reply:
Async are not executed in thread pool. It is will be executed concurrently through event loop. If you use execSync, it will block the event loop, however if you Async/ await then the event loop shall be in active and execute other pending actions.